.github/workflows/build-and-push-docker.yml

```diff
@@ -105,53 +75,72 @@ jobs:
-            docker push ${{ env.DOCKER_REPO_BROKER_SERVICE }}:latest
-            docker push ${{ env.DOCKER_REPO_BROKER_SERVICE }}:${{ github.sha }}
-          else
-            echo "Broker Service not in changed packages."
-          fi
-
-      - name: Build & Push Proof Service Image
+      - name: Build & Push Docker Images
+        id: build
         run: |
-          echo "Filtered Packages: ${{ needs.detect-changes.outputs.filtered_packages }}"
-          if [[ "${{ needs.detect-changes.outputs.filtered_packages }}" == *"3"* ]]; then
-            echo "Building and pushing Proof Service..."
-            echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_ACCESS_TOKEN }}" > npm-token
-            docker build -t ${{ env.DOCKER_REPO_PROOF_SERVICE }}:latest \
-              -f packages/proof-service/Dockerfile \
-              --platform=linux/amd64 \
-              --secret id=npm,src=npm-token .
-            docker tag ${{ env.DOCKER_REPO_PROOF_SERVICE }}:latest ${{ env.DOCKER_REPO_PROOF_SERVICE }}:${{ github.sha }}
-            docker push ${{ env.DOCKER_REPO_PROOF_SERVICE }}:latest
-            docker push ${{ env.DOCKER_REPO_PROOF_SERVICE }}:${{ github.sha }}
-          else
-            echo "Proof Service not in changed packages."
-          fi
-
-      - name: Build & Push Serverless Service Image
+          SERVICES=("serverless" "broker" "proof" "ftp")
+          DOCKER_REPOS=(
+            "orochinetwork/zkdb-serverless"
+            "orochinetwork/zkdb-broker-service"
+            "orochinetwork/zkdb-proof-service"
+            "orochinetwork/zkdb-ftp"
+          )
+          CHANGED_SERVICES=""
+          for i in ${!SERVICES[@]}; do
+            SERVICE="${SERVICES[$i]}"
+            DOCKER_REPO="${DOCKER_REPOS[$i]}"
+            NUM=$((i + 1))
+            if [[ -n "$DOCKER_REPO" && "${{ needs.detect-changes.outputs.filtered_packages }}" == *"$NUM"* ]]; then
+              echo "Building and pushing $SERVICE..."
+              echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_ACCESS_TOKEN }}" > npm-token 
+
+              Build the Docker image with a valid tag
+              docker build -t "$DOCKER_REPO:latest" \
+                -f "packages/$SERVICE/Dockerfile" \
+                --secret id=npm,src=./npm-token \
+                --platform=linux/amd64 .
+              
+              docker push "$DOCKER_REPO:latest"
+              docker tag "$DOCKER_REPO:latest" "$DOCKER_REPO:${{ env.version }}"
+              docker push "$DOCKER_REPO:${{ env.version }}"
+
+              CHANGED_SERVICES="$CHANGED_SERVICES $SERVICE"
+            else
+              echo "$SERVICE not changed or invalid Docker repo, skipping."
+            fi
+          done
+          echo "::set-output name=changed_services::$CHANGED_SERVICES"
+
+  gitops-push:
+    name: Push Changes to GitOps
+    needs: build
+    runs-on: ubuntu-latest
+    env:
+      version: ${{ github.sha }}
+      gitopsbranch: devops/test-gitops
```

Comment:
Why we need this branch?

Reply:
This branch is used in `test-argoCD` repo for argoCD to auto apply new changes. The purpose creating this is when we have new commit in main at `zkDatabase`, the commit ID will also be the tag of the image, and this commitID will be push in to this branch, so we keep the changes in a separate branch, not affecting other branch in `test-argoCD` repo. If we have other modification in helm chart, we simply merge these changes into this branch. 
Moreover, it's only a test branch, I will discuss with you about its name when my PR is merged.

Reply:
I think we should stick with `main` branch.